velox/vector/tests/VectorTest.cpp

```diff
@@ -2598,15 +2598,122 @@ TEST_F(VectorTest, rowCopyRanges) {
+    EXPECT_TRUE(dictionary->containsNullAt(i));
+  }
+
+  for (auto i = 1; i < dictionary->size(); i += 2) {
+    EXPECT_FALSE(dictionary->containsNullAt(i));
+  }
+
+  data = makeNullableFlatVector<int32_t>({1, std::nullopt, 3});
+  EXPECT_FALSE(data->containsNullAt(0));
+  EXPECT_TRUE(data->containsNullAt(1));
+  EXPECT_FALSE(data->containsNullAt(2));
+
+  dictionary = wrapInDictionary(indices, data);
+  EXPECT_FALSE(dictionary->containsNullAt(0));
+  EXPECT_FALSE(dictionary->containsNullAt(1));
+  EXPECT_TRUE(dictionary->containsNullAt(2));
+  EXPECT_TRUE(dictionary->containsNullAt(3));
+  EXPECT_FALSE(dictionary->containsNullAt(4));
+  EXPECT_FALSE(dictionary->containsNullAt(5));
+
+  dictionary = BaseVector::wrapInDictionary(nulls, indices, 6, data);
+  EXPECT_TRUE(dictionary->containsNullAt(0));
+  EXPECT_FALSE(dictionary->containsNullAt(1));
+  EXPECT_TRUE(dictionary->containsNullAt(2));
+  EXPECT_TRUE(dictionary->containsNullAt(3));
+  EXPECT_TRUE(dictionary->containsNullAt(4));
+  EXPECT_FALSE(dictionary->containsNullAt(5));
+
+  data = makeConstant(10, 3);
+  for (auto i = 0; i < data->size(); ++i) {
+    EXPECT_FALSE(data->containsNullAt(i));
+  }
+
+  data = makeNullConstant(TypeKind::INTEGER, 3);
+  for (auto i = 0; i < data->size(); ++i) {
+    EXPECT_TRUE(data->containsNullAt(i));
+  }
+}
+
+TEST_F(VectorTest, containsNullAtArrays) {
+  auto data = makeNullableArrayVector<int32_t>({
+      {{1, 2}},
+      {{1, 2, std::nullopt, 3}},
+      {{}},
+      std::nullopt,
+      {{1, 2, 3, 4}},
+  });
+
+  EXPECT_FALSE(data->containsNullAt(0));
+  EXPECT_TRUE(data->containsNullAt(1));
+  EXPECT_FALSE(data->containsNullAt(2));
+  EXPECT_TRUE(data->containsNullAt(3));
+  EXPECT_FALSE(data->containsNullAt(4));
+}
+
+TEST_F(VectorTest, containsNullAtMaps) {
+  auto data = makeNullableMapVector<int32_t, int64_t>({
+      {{{1, 10}, {2, 20}}},
+      {{{3, 30}}},
+      {{{1, 10}, {2, 20}, {3, std::nullopt}, {4, 40}}},
+      {{}},
+      std::nullopt,
+      {{{1, 10}, {2, 20}, {3, 30}, {4, 40}}},
+  });
+
+  EXPECT_FALSE(data->containsNullAt(0));
+  EXPECT_FALSE(data->containsNullAt(1));
+  EXPECT_TRUE(data->containsNullAt(2));
+  EXPECT_FALSE(data->containsNullAt(3));
+  EXPECT_TRUE(data->containsNullAt(4));
+  EXPECT_FALSE(data->containsNullAt(5));
+}
+
+TEST_F(VectorTest, containsNullAtStructs) {
+  auto data = makeRowVector(
+      {
+          makeNullableFlatVector<int32_t>({1, 2, std::nullopt, 4, 5}),
+          makeNullableFlatVector<int64_t>(
+              {10, std::nullopt, std::nullopt, 40, 50}),
+          makeNullableFlatVector<int16_t>({1, 2, 3, 4, 5}),
+          makeNullableFlatVector<int16_t>({10, 20, 30, 40, 50}),
+      },
+      [](auto row) { return row == 3; });
+
+  EXPECT_FALSE(data->containsNullAt(0));
+  EXPECT_TRUE(data->containsNullAt(1));
+  EXPECT_TRUE(data->containsNullAt(2));
+  EXPECT_TRUE(data->containsNullAt(3));
+  EXPECT_FALSE(data->containsNullAt(4));
```

Comment:
Should we also have a test for nested containers?

Reply:
Sure. Let me add some.

Reply:
@pedroerp Added.